src/pages/Browse/index.js

```diff
@@ -1,11 +1,166 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { fetchAllUserRecords } from '../../store/record/actions'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+import { selectToken } from '../../store/user/selectors'
+import {
+	selectUserRecords,
+	selectCountOfUserRecords,
+} from '../../store/record/selectors'
+import { getSelectOptions } from '../../functions'
+import Loading from '../../components/Loading'
+import RecordCard from '../../components/RecordCard'
+import '../../components/RecordCard/RecordCard.css'
+import TextField from '@material-ui/core/TextField'
+import MultipleSelect from '../../components/MultipleSelect'
+import Button from '@material-ui/core/Button'
 
 const Browse = () => {
-  return (
-    <div>
-      Browse
-    </div>
-  )
+	const dispatch = useDispatch()
+	const token = useSelector(selectToken)
+	const history = useHistory()
+	const records = useSelector(selectUserRecords)
+	const count = useSelector(selectCountOfUserRecords)
+	const [genre, setGenre] = useState([])
+	const [style, setStyle] = useState([])
+	const [year, setYear] = useState([])
+	const [artist, setArtist] = useState([])
+	const [title, setTitle] = useState('')
+
+	useEffect(() => {
+		if (records.length === 0) {
+			dispatch(fetchAllUserRecords())
+		}
+	}, [])
+
+	if (!token) {
+		history.push('/')
+	}
+	if (!records) {
+		return <Loading />
+	}
+
+	// const updatePagination = () => {
+	// 	dispatch(fetchPageUserRecords(records.length, 10))
+	// }
+
+	const getOptionsRecords = getSelectOptions(records)
+	const genreArray = getOptionsRecords('genre')
+	const styleArray = getOptionsRecords('style')
+	const yearArray = getOptionsRecords('year')
+	const artistArray = getOptionsRecords('artist')
+
+	const updateFilter = (key, selection) => {
+		if (key === 'genre') {
+			setGenre(selection)
+		}
+		if (key === 'style') {
+			setStyle(selection)
+		}
+		if (key === 'year') {
+			setYear(selection)
+		}
+		if (key === 'artist') {
+			setArtist(selection)
+		}
+	}
+
+	const filteredRecords = records.filter((record) => {
+		if (
+			(genre.some((item) => record.genre.includes(item)) ||
+				genre.length === 0) &&
+			(style.some((item) => record.style.includes(item)) ||
+				style.length === 0) &&
+			(year.some((item) => record.year === parseInt(item)) ||
+				year.length === 0) &&
+			(artist.some((item) => record.artist.includes(item)) ||
+				artist.length === 0) &&
+			(record.title.toLowerCase().includes(title.toLowerCase()) || !title)
+		) {
+			return true
+		}
+		return false
+	})
```

Comment:
I would move this to the `functions` file, also, we call that file `helpers.js` or `utils.js` or (`/utils/index.js ofc) but you were exactly in the right track!

Reply:
One easy improvement is just getting rid of the `if` and just returning that which you have in the condition. It's the same, if it resolves to true you'll return true inside the if.